src/codeu/chat/server/Controller.java

```diff
@@ -179,6 +179,42 @@ public void newConversationSubscription(String title, Uuid user) {
   }
 
   @Override
+  public void addMember(String user, Uuid conversation) {
+    final User u = model.userByText().first(user);
+    final ConversationHeader conversationHeader = model.conversationById().first(conversation);
+
+    AccessLevel level = conversationHeader.getAccessLevel(u);
+
+    if(level.hasMemberAccess()){
+      System.out.println("The user is already a member.");
+    }
+    else if(level.hasOwnerAccess() || level.hasCreatorAccess()){
+      System.out.println("You cannot lower an owner's or creator's status.");
+    }
+    else{
+      level.setMemberStatus();
+    }
+  }
+
+  @Override
+  public void addOwner(String user, Uuid conversation) {
+    final User u = model.userByText().first(user);
+    final ConversationHeader conversationHeader = model.conversationById().first(conversation);
+
+    AccessLevel level = conversationHeader.getAccessLevel(u);
+
+    if(level.hasCreatorAccess()){
```

Comment:
It seems from the prints that this was supposed to be .hasOwnerAccess(), otherwise these two if's would be checking for the same condition